roles/k3s/master/tasks/main.yml

```diff
@@ -64,7 +64,8 @@
     cmd: "systemd-run -p RestartSec=2 \
                       -p Restart=on-failure \
                       --unit=k3s-init \
-                      k3s server {{ server_init_args }}"
+                      k3s server {{ server_init_args }} \
+                      {{ '--node-taint CriticalAddonsOnly=true:NoExecute' if k3s_single_node|bool == false else ''}}"
```

Comment:
Suggestion: How about specifying the taints via Ansible host vars? This would make our handling of taints more flexible and more transparent to the user :-)

Reply:
My only concern with that is then we have to make the tolerations within kubevip and metallb dynamic too which to me seems like it could create problems

Reply:
Both great points.  I do think that this is fine to apply these specific taints based on the number of nodes.  

I think that's the general expectation of those using k3s.

* servers have taints to repel general workloads
* agents do not
* if only one node, we should not apply the taint so that general workloads can run on it (becomes server + agent)
* critical add-ons like `kube-vip` and `metal-lb` should run anywhere (servers & agents)

At least that's my thinking

Reply:
Thats how I thought about it too, generally just going on ranchers documentation for k3s the expectation of control plane/etcd nodes having that specific taint seemed pretty standard.

Reply:
I actually have some cases planned where I would like to subvert those expectations 😅 So when I'll get to implementing it, I will check if I can find a way to add a bit more flexibility here without increasing the complexity too much :-)